pkg/cmd/gist/list/list.go

```diff
@@ -33,17 +37,46 @@ func NewCmdList(f *cmdutil.Factory, runF func(*ListOptions) error) *cobra.Comman
 
 	var flagPublic bool
 	var flagSecret bool
+	var flagFilter string
 
 	cmd := &cobra.Command{
-		Use:     "list",
-		Short:   "List your gists",
+		Use:   "list",
+		Short: "List your gists",
+		Long: heredoc.Docf(`
+			List gists from your user account.
+
+			You can use a regular expression to filter the description, file names,
+			or even the content of files in the gist. See https://pkg.go.dev/regexp/syntax
+			for the regular expression syntax you can pass to %[1]s--filter%[1]s. Pass
+			%[1]s--include-content%[1]s to also search the content of files noting that
+			this will take longer since all files' content is fetched.
+		`, "`"),
+		Example: heredoc.Doc(`
+			# list all secret gists from your user account
+			$ gh gist list --secret
+
+			# find all gists from your user account mentioning "octo" anywhere
+			$ gh gist list --filter octo --include-content
+		`),
 		Aliases: []string{"ls"},
 		Args:    cobra.NoArgs,
 		RunE: func(cmd *cobra.Command, args []string) error {
 			if opts.Limit < 1 {
 				return cmdutil.FlagErrorf("invalid limit: %v", opts.Limit)
 			}
 
+			if flagFilter == "" {
+				if opts.IncludeContent {
+					return cmdutil.FlagErrorf("cannot use --include-content without --filter")
+				}
+			}
+
+			if filter, err := regexp.CompilePOSIX(flagFilter); err != nil {
+				return err
+			} else {
+				opts.Filter = filter
+			}
```

Comment:
Minor simplify nit

```suggestion
			if filter, err := regexp.CompilePOSIX(flagFilter); err != nil {
				return err
			}
			opts.Filter = filter
```

Reply:
That filter would be undefined since it was created in the conditional scope.